File: backend/api/schemas/service.py
from pydantic import BaseModel, Field


class ServiceCreate(BaseModel):
    name: str
    description: str | None = None
    position: int = Field(default=0, ge=0)
    is_hidden: bool = False


class ServiceUpdate(BaseModel):
    name: str
    description: str | None = None
    position: int = Field(default=0, ge=0)
    is_hidden: bool = False


class ServiceResponse(BaseModel):
    id: int
    name: str
    description: str | None = None
    position: int
    is_hidden: bool

File: frontend/src/types/service.ts
import type { PublicPriceResponse } from './price'

export interface ServiceCreate {
  name: string
  description?: string | null
  position?: number
  is_hidden?: boolean
}

export interface ServiceUpdate {
  name: string
  description?: string | null
  position?: number
  is_hidden?: boolean
}

export interface ServiceResponse {
  id: number
  name: string
  description: string | null
  position: number
  is_hidden: boolean
}

export interface PublicServiceResponse {
  id: number
  name: string
  description: string | null
  position: number
  prices: PublicPriceResponse[]
}